src/binary-reader-objdump.c

```diff
@@ -370,14 +372,16 @@ static WasmResult on_function_signature(uint32_t index,
                                         void* user_data) {
   Context* ctx = user_data;
 
-  print_details(user_data, " - func[%d] sig=%d\n", ctx->function_index, sig_index);
-  ctx->function_index++;
+  print_details(user_data, " - func[%d] sig=%d\n",
+                ctx->num_func_imports + index, sig_index);
   return WASM_OK;
 }
 
 static WasmResult begin_function_body(uint32_t index, void* user_data) {
   Context* ctx = user_data;
 
+  index += ctx->num_func_imports;
```

Comment:
Would it make sense to change the index passed into begin_function_body (and end_function_body, I suppose) to be the function-index-space index? From a quick scan, it seems like that's what most of the places that use that argument ultimately want.

The one thing that a quick scan of the code turns up that needs the function-section index is this, in binary-reader-interpreter.c's begin_function_body:
```
  Uint32Vector* fixups = &ctx->func_fixups.data[index];
```
but if that's all, then perhaps it could be handled by adding a function like `get_func_by_env_index` but for reading fixups.